apps/studio/src/components/sidebar/ConnectionSidebar.vue

```diff
@@ -149,6 +184,18 @@ const log = rawLog.scope('connection-sidebar');
       }
     }),
     watch: {
+      // If we load with some pins, this will reinitialize split to reflect that
+      noPins(value) {
+        if (!value)
+          this.split = this.getSplit()
+      },
+      // Check if we need to reinitialize split based on pinnedConnections length
+      pinnedConnections(value) {
+        if (value.length < 2) {
+          this.split?.destroy()
```

Comment:
if you always need to call destroy before setting a new split, change the method to `buildSplit` and do it in there. See below comments.